bundles/AdminBundle/Resources/public/js/pimcore/asset/asset.js

```diff
@@ -41,10 +41,8 @@ pimcore.asset.asset = Class.create(pimcore.element.abstract, {
                 throw "asset is locked";
             }
 
-            if(this.type !== this.data.type) {
-                pimcore.helpers.closeAsset(this.id);
-                pimcore.helpers.openAsset(this.id, this.data.type);
-                return;
+            if (this.type !== this.data.type) {
```

Comment:
@dvesh3 Should we really change the type on the fly? As `type` is kind of a static variable that is defined by the actual implementation, see e.g.: 
https://github.com/pimcore/pimcore/blob/10.x/bundles/AdminBundle/Resources/public/js/pimcore/asset/audio.js#L21

Actually if the type in the UI doesn't match with the one sent from the server, there's something pretty wrong ... but maybe we should show an error message instead, that recommends to reload the admin ui? wdyt?

Reply:
yes, I think a warning to load the admin UI is a better idea than silently changing the type on the fly. This could lead to unwanted behavior in admin interface, if the response is not correct from the server for some reason.

Good suggestion! I'll update it 😉

Reply:
However, we have to fix the type for zip archives response in the server as it returns unknown at the moment and then will end up with a warning every time you upload the zip file.

Reply:
Added archive js type(to be consistent with models) and a warning message when the asset type is changed in the server.